python/hail/expr/expressions/base_expression.py

```diff
@@ -365,18 +365,26 @@ def _promote_scalar(self, typ):
             return hail.float64(self)
 
     def _promote_numeric(self, typ):
-        if isinstance(typ, tarray):
-            if isinstance(self.dtype, tarray):
-                return hail.map(lambda x: x._promote_scalar(typ.element_type), self)
-            else:
-                return self._promote_scalar(typ.element_type)
-        elif isinstance(self, expressions.Aggregable):
```

Comment:
the aggregable logic has moved to a new agg_expr typechecker (which simplifies lots of stuff)